src/wasm/wasm-type.cpp

```diff
@@ -594,8 +600,9 @@ bool Type::isSubType(Type left, Type right) {
       return true;
     }
     // Various things are subtypes of eqref.
-    if ((left == Type::i31ref || left.getHeapType().isArray() ||
-         left.getHeapType().isStruct()) &&
+    auto leftHeap = left.getHeapType();
+    if ((leftHeap == HeapType::i31 || leftHeap.isArray() ||
+         leftHeap.isStruct()) &&
```

Comment:
I assume `eqref` is short for `(ref null eq)` and that there is also a `(ref eq)`. In that case this code probably should compare with `rightHeap == HeapType::eq` rather than with `right == Type::eqref`. It will have to handle differing nullability between left and right as well.

Reply:
Thanks, yes. Pushed a fix now.